packages/litexa/src/parser/handler.coffee

```diff
@@ -443,7 +443,15 @@ handlerSteps.createFinalResult = (stateContext) ->
 
   # build outputSpeech and reprompt from the accumulators
   joinSpeech = (arr, language = 'default') ->
-    result = arr.join(' ')
+    return '' unless arr
+    result = arr[0]
+    for line in arr[1..]
+      # If the line starts with punctuation, don't add a space before.
+      if line.match /^[?!:;,.]/
+        result += line
+      else
+        result += " #{line}"
+
     result = result.replace /(  )/g, ' '
```

Comment:
Could remove the parentheses from the regex, not needed.

Reply:
Looks like that doesn't work with Coffescript syntax, which may be why they're there.